GPU/Vulkan/FramebufferVulkan.cpp

```diff
@@ -626,4 +632,74 @@ void FramebufferManagerVulkan::Resized() {
 	if (UpdateSize()) {
 		DestroyAllFBOs();
 	}
+
+	// Might have a new post shader - let's compile it.
+	CompilePostShader();
 }
+
+void FramebufferManagerVulkan::CompilePostShader() {
+	if (postVs_) {
+		vulkan_->Delete().QueueDeleteShaderModule(postVs_);
+		postVs_ = nullptr;
+	}
+	if (postFs_) {
+		vulkan_->Delete().QueueDeleteShaderModule(postFs_);
+		postFs_ = nullptr;
+	}
+
+	const ShaderInfo *shaderInfo = nullptr;
+	if (g_Config.sPostShaderName == "Off") {
+		usePostShader_ = false;
+		return;
+	}
+
+	usePostShader_ = false;
+
+	ReloadAllPostShaderInfo();
+	shaderInfo = GetPostShaderInfo(g_Config.sPostShaderName);
+	std::string errorVSX, errorFSX;
+	std::string vsSource;
+	std::string fsSource;
+	if (shaderInfo) {
+		postShaderAtOutputResolution_ = shaderInfo->outputResolution;
+		size_t sz;
+		char *vs = (char *)VFSReadFile(shaderInfo->vertexShaderFile.c_str(), &sz);
+		if (!vs)
+			return;
+		char *fs = (char *)VFSReadFile(shaderInfo->fragmentShaderFile.c_str(), &sz);
+		if (!fs) {
+			free(vs);
+			return;
+		}
+		std::string vsSourceGLSL = vs;
+		std::string fsSourceGLSL = fs;
+		free(vs);
+		free(fs);
+		TranslatedShaderMetadata metaVS, metaFS;
+		if (!TranslateShader(&vsSource, GLSL_VULKAN, &metaVS, vsSourceGLSL, GLSL_140, Draw::ShaderStage::VERTEX, &errorVSX))
+			return;
+		if (!TranslateShader(&fsSource, GLSL_VULKAN, &metaFS, fsSourceGLSL, GLSL_140, Draw::ShaderStage::FRAGMENT, &errorFSX))
+			return;
+	} else {
+		return;
+	}
+	I18NCategory *gr = GetI18NCategory("Graphics");
+
+	// TODO: Delete the old pipeline?
+
+	std::string errorVS;
+	std::string errorFS;
+	postVs_ = CompileShaderModule(vulkan_, VK_SHADER_STAGE_VERTEX_BIT, vsSource.c_str(), &errorVS);
+	postFs_ = CompileShaderModule(vulkan_, VK_SHADER_STAGE_FRAGMENT_BIT, fsSource.c_str(), &errorFS);
+
+	VkRenderPass backbufferRP = (VkRenderPass)draw_->GetNativeObject(Draw::NativeObject::BACKBUFFER_RENDERPASS);
+
+	if (postVs_ && postFs_) {
+		pipelinePostShader_ = vulkan2D_->GetPipeline(backbufferRP, postVs_, postFs_, true, Vulkan2D::VK2DDepthStencilMode::NONE);
+	} else {
+		ELOG("Failed to compile.");
```

Comment:
nit: osm?

-[Unknown]

Reply:
It's a bit of a misnomer (should be "Failed to create pipeline for post proc" or something) but this shouldn't really happen, it's much more common that the actual shaders fail to compile which is checked for earlier. Though those should OSM indeed.